lib/next_ls/snippet.ex

```diff
@@ -171,7 +182,47 @@ defmodule NextLS.Snippet do
     }
   end
 
-  def get(_label, _trigger_character) do
+  def get(_label, _trigger_character, _opts) do
     nil
   end
+
+  defp infer_module_name(uri) do
+    result =
+      uri
+      |> Path.split()
+      |> Enum.reduce(false, fn
```

Comment:
Can you explain what the first element of the accumulator signifies here?

The clause with support is confusing me a tad.

Reply:
Usually the module in `test/support` don't use `Test` in the module name like https://github.com/elixir-tools/next-ls/blob/main/test/support/utils.ex

> Can you explain what the first element of the accumulator signifies here?

do you mean `false`?

Reply:
I meant the tuple `{:test, []}`. I'm not sure what the atom is doing

Reply:
It's used [here](https://github.com/elixir-tools/next-ls/pull/398/files#diff-58f5082ef38a94ce607842d596a0ec1d974224124c9648642c2d46f5435ce750R200) to check if there is `support` under `test`

Reply:
Oh my bad, I'll try to explain.

I'm looking for three folders

- `lib` where usually the code is, `lib/foo.ex` it's translated to `Foo` (without `lib`)
- `test` where usually the tests are, `test/foo_test.exs` it's translated to `FooTest` (without `test`)
- `support` where usually are utils for tests, `test/support/foo.ex` it's translated to `Foo` (without `test` and `support`)

here are some tests that can help https://github.com/elixir-tools/next-ls/blob/9d65b759ac0cf86c943190342cf829097fc62346/test/next_ls/snippet_test.exs

Reply:
oh, i understand now, thanks!

basically you are just striping `lib/`, `test/` and `test/support/`  from the beginning for the path.

Reply:
Yes 👍

Reply:
@lucacervello When you use someone else's code, please mention the source. If possible, contacting the author would be best; otherwise, it appears quite unethical.

https://github.com/lexical-lsp/lexical/blob/2178b4beab5d1a04cb83990e09b3dc232ed941cb/apps/server/lib/lexical/server/code_intelligence/completion/translations/macro.ex#L567C5-L603C8

Reply:
I've removed the code on main.

Sorry about that!